File: Python/underautomation/universal_robots/primary_interface/joint_data.py
import typing
from underautomation.universal_robots.common.joint_modes import JointModes
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import JointData as joint_data

class JointData:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = joint_data()
		else:
			self._instance = _internal
	@property
	def position(self) -> float:
		return self._instance.Position
	@position.setter
	def position(self, value: float):
		self._instance.Position = value
	@property
	def target_position(self) -> float:
		return self._instance.TargetPosition
	@target_position.setter
	def target_position(self, value: float):
		self._instance.TargetPosition = value
	@property
	def actual_speed(self) -> float:
		return self._instance.ActualSpeed
	@actual_speed.setter
	def actual_speed(self, value: float):
		self._instance.ActualSpeed = value
	@property
	def current(self) -> float:
		return self._instance.Current
	@current.setter
	def current(self, value: float):
		self._instance.Current = value
	@property
	def voltage(self) -> float:
		return self._instance.Voltage
	@voltage.setter
	def voltage(self, value: float):
		self._instance.Voltage = value
	@property
	def temperature(self) -> float:
		return self._instance.Temperature
	@temperature.setter
	def temperature(self, value: float):
		self._instance.Temperature = value
	@property
	def joint_mode(self) -> JointModes:
		return JointModes(self._instance.JointMode)
	@joint_mode.setter
	def joint_mode(self, value: JointModes):
		self._instance.JointMode = value

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/JointData.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace PrimaryInterface {
	/// <summary>
	/// Joint data
	/// </summary>
	public class JointData {


		public override string ToString()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public override bool Equals(object obj)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public override int GetHashCode()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public JointData()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Angular joint position in radian
		/// </summary>
		[PackageDescription("Angular joint position in radian", PackageUnit.Radian)]
public double Position { get; set; }

		/// <summary>
		/// Angular target position in radian
		/// </summary>
		[PackageDescription("Angular target position in radian", PackageUnit.Radian)]
public double TargetPosition { get; set; }

		/// <summary>
		/// Joint rotation speed in rad/s
		/// </summary>
		[PackageDescription("Joint rotation speed in rad/s", PackageUnit.RadianPerSecond)]
public double ActualSpeed { get; set; }

		/// <summary>
		/// Motor current in Amps
		/// </summary>
		[PackageDescription("Motor current in Amps", PackageUnit.Amp)]
public float Current { get; set; }

		/// <summary>
		/// Motor voltage in Volts
		/// </summary>
		[PackageDescription("Motor voltage in Volts", PackageUnit.Volt)]
public float Voltage { get; set; }

		/// <summary>
		/// Joint temperature in °C
		/// </summary>
		[PackageDescription("Joint temperature in °C", PackageUnit.CelsiusDegree)]
public float Temperature { get; set; }

		/// <summary>
		/// Joint mode
		/// </summary>
		[PackageDescription("Joint mode")]
public JointModes JointMode { get; set; }
	}
}
